crates/cli/src/templates/binding.java

```diff
@@ -0,0 +1,62 @@
+package PARSER_NS_CLEANED.jtreesitter.LOWER_PARSER_NAME;
+
+import java.lang.foreign.*;
+
+public final class PARSER_CLASS_NAME {
+    private static final ValueLayout VOID_PTR =
+            ValueLayout.ADDRESS.withTargetLayout(MemoryLayout.sequenceLayout(Long.MAX_VALUE, ValueLayout.JAVA_BYTE));
+    private static final FunctionDescriptor FUNC_DESC = FunctionDescriptor.of(VOID_PTR);
+    private static final Linker LINKER = Linker.nativeLinker();
+    private static final TreeSitterJava INSTANCE = new PARSER_CLASS_NAME();
+
+    private final Arena arena = Arena.ofAuto();
+    private SymbolLookup lookup = null;
+
+    /**
+     * Get the tree-sitter language for this grammar.
+     */
+    public static MemorySegment language() {
+        if (INSTANCE.lookup == null)
+            INSTANCE.lookup = INSTANCE.findLibrary();
+        return language(INSTANCE.lookup);
+    }
+
+    /**
+     * Get the tree-sitter language for this grammar.
+     *
+     * <strong>The {@linkplain Arena} used in the {@code lookup}
+     * must not be closed while the language is being used.</strong>
+     */
+    public static MemorySegment language(SymbolLookup lookup) {
+        return call(lookup, "tree_sitter_PARSER_NAME");
+    }
+
+    private SymbolLookup findLibrary() {
+        try {
+            var library = System.mapLibraryName("tree-sitter-KEBAB_PARSER_NAME");
+            return SymbolLookup.libraryLookup(library, arena);
+        } catch (IllegalArgumentException ex1) {
+            try {
+                return SymbolLookup.loaderLookup();
+            } catch (UnsatisfiedLinkError ex2) {
```

Comment:
Is this correct? At least based on the documentation `loaderLookup()` won't throw `UnsatisfiedLinkError`.